src/sfizz/SfzFilter.cpp

```diff
@@ -0,0 +1,256 @@
+};
+
+template <unsigned NCh>
+Filter<NCh>::Filter()
+    : P{new Impl}
+{
+}
+
+template <unsigned NCh>
+Filter<NCh>::~Filter()
+{
+}
+
+template <unsigned NCh>
+void Filter<NCh>::init(double sampleRate)
+{
+    P->fDspLpf1p.init(sampleRate);
+    P->fDspLpf2p.init(sampleRate);
+    P->fDspLpf4p.init(sampleRate);
+    P->fDspLpf6p.init(sampleRate);
+    P->fDspHpf1p.init(sampleRate);
+    P->fDspHpf2p.init(sampleRate);
+    P->fDspHpf4p.init(sampleRate);
+    P->fDspHpf6p.init(sampleRate);
+    P->fDspBpf1p.init(sampleRate);
+    P->fDspBpf2p.init(sampleRate);
+    P->fDspBpf4p.init(sampleRate);
+    P->fDspBpf6p.init(sampleRate);
+    P->fDspApf1p.init(sampleRate);
+    P->fDspBrf1p.init(sampleRate);
+    P->fDspBrf2p.init(sampleRate);
+    P->fDspPink.init(sampleRate);
+    P->fDspLpf2pSv.init(sampleRate);
+    P->fDspHpf2pSv.init(sampleRate);
+    P->fDspBpf2pSv.init(sampleRate);
+    P->fDspBrf2pSv.init(sampleRate);
+    P->fDspLsh.init(sampleRate);
+    P->fDspHsh.init(sampleRate);
+    P->fDspPeq.init(sampleRate);
+}
+
+template <unsigned NCh>
+void Filter<NCh>::clear()
+{
+    switch (P->fType) {
+    case kFilterNone: break;
+    case kFilterApf1p: P->fDspApf1p.instanceClear(); break;
+    case kFilterBpf1p: P->fDspBpf1p.instanceClear(); break;
+    case kFilterBpf2p: P->fDspBpf2p.instanceClear(); break;
+    case kFilterBpf4p: P->fDspBpf4p.instanceClear(); break;
+    case kFilterBpf6p: P->fDspBpf6p.instanceClear(); break;
+    case kFilterBrf1p: P->fDspBrf1p.instanceClear(); break;
+    case kFilterBrf2p: P->fDspBrf2p.instanceClear(); break;
+    case kFilterHpf1p: P->fDspHpf1p.instanceClear(); break;
+    case kFilterHpf2p: P->fDspHpf2p.instanceClear(); break;
+    case kFilterHpf4p: P->fDspHpf4p.instanceClear(); break;
+    case kFilterHpf6p: P->fDspHpf6p.instanceClear(); break;
+    case kFilterLpf1p: P->fDspLpf1p.instanceClear(); break;
+    case kFilterLpf2p: P->fDspLpf2p.instanceClear(); break;
+    case kFilterLpf4p: P->fDspLpf4p.instanceClear(); break;
+    case kFilterLpf6p: P->fDspLpf6p.instanceClear(); break;
+    case kFilterPink: P->fDspPink.instanceClear(); break;
+    case kFilterLpf2pSv: P->fDspLpf2pSv.instanceClear(); break;
+    case kFilterHpf2pSv: P->fDspHpf2pSv.instanceClear(); break;
+    case kFilterBpf2pSv: P->fDspBpf2pSv.instanceClear(); break;
+    case kFilterBrf2pSv: P->fDspBrf2pSv.instanceClear(); break;
+    case kFilterLsh: P->fDspLsh.instanceClear(); break;
+    case kFilterHsh: P->fDspHsh.instanceClear(); break;
+    case kFilterPeq: P->fDspPeq.instanceClear(); break;
+    }
+}
+
+template <unsigned NCh>
+template <class F>
+void Filter<NCh>::Impl::process(F &filter, const float *const in[NCh], float *const out[NCh], float cutoff, float q, float pksh, unsigned nframes)
+{
+    filter.setCutoff(cutoff);
+    filter.setQ(q);
+    filter.setPkShGain(pksh);
+    filter.compute(nframes, const_cast<float **>(in), const_cast<float **>(out));
```

Comment:
I'm not a super big fan of the const casts... Is there a reason Faust generates the interfaces as `float**`?

Reply:
I'm afraid it's the way it is for a long time, unfortunately.
I don't see for it any good reason, it's a thing to deal with.